pkg/dipper/mapUtils.go

```diff
@@ -354,16 +346,20 @@ func mergeModifier(dst map[string]interface{}) {
 // MergeMap : merge the data from source to destination with some overriding rule
 func MergeMap(dst map[string]interface{}, src interface{}) map[string]interface{} {
 	dst = CombineMap(dst, src)
-	for k, v := range dst {
-		if k[len(k)-1] == '-' {
-			if ev, ok := dst[k[:len(k)-1]]; !ok || ev == nil {
-				dst[k[:len(k)-1]] = v
-			}
-			delete(dst, k)
-		}
-	}
+	overwriteDupKeys(dst)
 
 	mergeModifier(dst)
```

Comment:
We are already calling `overwriteDupKeys`  in `mergeModifier`, probably don't need to call `overwriteDupKeys` here any more.  It was probably redundant in my code to begin with.  After removing the redundancy, If this part of code show up only once, maybe we don't need a separate function for this.  But, either way, it is better than my original code. 👍

Reply:
ahh I see. Let me remove this part and revert the other chunk of code back. Otherwise, this would be very confusing for people to look at because this particular function got invoke twice in a row